packages/core/utils/src/config.js

```diff
@@ -32,15 +36,21 @@ export async function resolveConfig(
 
   for (const filename of filenames) {
     let file = path.join(filepath, filename);
-    if ((await fs.exists(file)) && (await fs.stat(file)).isFile()) {
-      return file;
+    try {
+      if (existingFiles.has(file)) return file;
+
+      if ((await fs.stat(file)).isFile()) {
+        existingFiles.add(file);
+        return file;
+      }
+    } catch {
+      // empty
```

Comment:
Instead of using a set, you could use a map. That way, you can also cache what files *don't* exist in addition to those that do.